core/src/main/java/de/danoeh/antennapod/core/service/download/DownloadService.java

```diff
@@ -205,9 +205,10 @@ public void onDestroy() {
         Log.d(TAG, "Service shutting down");
         isRunning = false;
 
+        boolean showAutoDownloadReport = UserPreferences.showAutoDownloadReport();
         if (ClientConfig.downloadServiceCallbacks.shouldCreateReport()
-                && UserPreferences.showDownloadReport()) {
-            notificationManager.updateReport(reportQueue);
+                && (UserPreferences.showDownloadReport() || showAutoDownloadReport)) {
+            notificationManager.updateReport(reportQueue, showAutoDownloadReport);
```

Comment:
Just wondering... The two kinds of notifications are now quite different (Channel, text, maybe even icon). Would it make sense to extract the auto download notification to a new class instead of handling everything in the existing `notificationManager`? I have not checked if this will lead to a lot of code duplication, though.

Reply:
I'm not 100% sure what you mean here. `DownloadService` seems to be the right place to kick off a notification and `notificationManager` (to me anyway) seems like a class that will generate whatever notification is required for a given set of parameters. IMO if we did create another class there would be a fair bit of duplication.